pkg/envoy/lds/listener.go

```diff
@@ -200,3 +150,124 @@ func getCIDRRange(cidr string) (*xds_core.CidrRange, error) {
 
 	return cidrRange, nil
 }
+
+// getFilterForService builds a network filter action for traffic destined to a specific service
+func getFilterForService(dstSvc service.MeshService, cfg configurator.Configurator) (*xds_listener.Filter, error) {
+	var marshalledFilter *any.Any
+	var err error
+
+	marshalledFilter, err = envoy.MessageToAny(
+		getHTTPConnectionManager(route.OutboundRouteConfigName, cfg))
+	if err != nil {
+		log.Error().Err(err).Msgf("Error marshalling HTTPConnManager object")
+		return nil, err
+	}
+
+	return &xds_listener.Filter{
+		Name:       wellknown.HTTPConnectionManager,
+		ConfigType: &xds_listener.Filter_TypedConfig{TypedConfig: marshalledFilter},
+	}, nil
+}
+
+// getFilterChainMatchForService builds a filter chain to match the destination traffic.
+// Filter Chain currently match on destination IP for possible service endpoints
+func getFilterChainMatchForService(dstSvc service.MeshService, catalog catalog.MeshCataloger, cfg configurator.Configurator) (*xds_listener.FilterChainMatch, error) {
+	ret := &xds_listener.FilterChainMatch{}
+
+	endpoints, err := catalog.GetResolvableServiceEndpoints(dstSvc)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error getting GetResolvableServiceEndpoints for %s", dstSvc.String())
+		return nil, err
+	}
+
+	if len(endpoints) == 0 {
+		log.Error().Err(err).Msgf("No resolvable addresses retured for service %s", dstSvc.String())
+		return nil, errNoValidTargetEndpoints
+	}
+
+	for _, endp := range endpoints {
+		ret.PrefixRanges = append(ret.PrefixRanges, &xds_core.CidrRange{
+			AddressPrefix: endp.IP.String(),
+			PrefixLen: &wrapperspb.UInt32Value{
+				Value: singleIpv4Mask,
+			},
+		})
+	}
+
+	return ret, nil
+}
+
+func getFilterChains(catalog catalog.MeshCataloger, cfg configurator.Configurator, localSvc []service.MeshService) ([]*xds_listener.FilterChain, error) {
+	var ret []*xds_listener.FilterChain
+	var dstServicesSet map[service.MeshService]struct{} = make(map[service.MeshService]struct{}) // Set, avoid dups
+
+	outboundSvc, err := catalog.ListAllowedOutboundServices(localSvc[0])
```

Comment:
Since we are making an assumption around a proxy being associated with a single service (as is the case now), I'd reference issue https://github.com/openservicemesh/osm/issues/1575

Reply:
arf, I had the comment to begin with, I removed it as I thought I had too much chatter on the func. Readding.